src/app/routes/ssoRoutes.js

```diff
@@ -228,9 +228,13 @@ module.exports = (app) => {
         errType = 'MISSING_QUERY_PARAMS';
         throw 'some of the query params are missing';
       }
-      userName = req.query.id;
+      errType = 'DECRYPTING_DATA';
+      const decryptedData = decrypt(parseJson(decodeURIComponent(req.query.id)));
+      const userData = parseJson(decryptedData);
+      errType = 'VERIFY_TOKEN';
+      verifyToken(userData);
```

Comment:
@deveshMantra - why are we trying to reuse the same verifyToken method here also? It will be a problem when you need to change verifyToken method to add/change JWT validations. It will then break this functionality. All we need here is validating expiry timestamp of the decrypted string - it could be a separate small utility function.
@vinukumar-vs